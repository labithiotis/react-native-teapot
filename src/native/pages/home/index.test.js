import React from 'react';
import { shallow } from 'enzyme';

import { Home } from './index';

describe('Home', () => {
  let component;

  beforeEach(() => {
    const props = global.defaultProps({
      screenProps: {
        locale: {
          pages: {
            home: {
              title: 'Title',
              intro: 'The Intro',
              button: 'Click Me',
            },
          },
        },
      },
    });
    component = shallow(<Home {...props} />);
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
