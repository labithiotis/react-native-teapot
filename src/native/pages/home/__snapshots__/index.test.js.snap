// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#132d42",
        "flex": 1,
      },
      Object {
        "flex": 1,
        "padding": 20,
      },
    ]
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 40,
            "fontWeight": "bold",
          },
          Object {
            "color": "#FFAE00",
          },
        ]
      }
    >
      Title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#6FD8FF",
            "fontSize": 16,
          },
          Object {
            "color": "#FF7808",
          },
        ]
      }
    >
      The Intro
    </Text>
    <Icon
      allowFontScaling={false}
      name="globe"
      size={12}
      style={
        Object {
          "color": "hsl(206.79999999999995, 55.3%, 21.7%)",
          "fontSize": 100,
          "marginTop": 20,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      focusedOpacity={0.7}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "borderColor": "#ffffff",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 20,
              "marginTop": 20,
              "paddingBottom": 10,
              "paddingLeft": 30,
              "paddingRight": 30,
              "paddingTop": 10,
            },
            undefined,
          ]
        }
      >
        <Icon
          allowFontScaling={false}
          name="reddit-alien"
          size={12}
          style={
            Object {
              "fontSize": 20,
              "right": -4,
            }
          }
        />
          
        Click Me
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
</View>
`;
